nwbwidgets/controllers/basecontroller.py

```diff
@@ -0,0 +1,81 @@
+"""
+Base class definition of all controllers.
+
+Attempted to make the class abstract via `abc.ABC` to use `abc.abstractmethod` but ran into metaclass conflict
+issues with `ipywidgets.Box`. Undefined methods instead raise NotImplementedErrors.
+"""
+from typing import Dict, Any
+
+import ipywidgets
+
+
+class BaseController(ipywidgets.Box):
+    """
+    Base definition of all Controllers.
+
+    A Controller is a container of objects such as widgets, including other controllers, that exposes all important
+    components as non-private attributes at the outermost level for simplified reference.
+
+    This is in constrast to defining an ipywidget.Box of other Boxes, where the only way to reference a particular
+    sub-widget component is by navigating the children tree, knowing the set of levels and indices required to find
+    a particular child.
+    """
+
+    def __init__(self):
+        """
+        Defines the workflow for initializing a Controller.
+
+        Overriding this method in a child class without calling super().__init__() is highly discouraged.
+
+        To align the children vertically, set `your_controller.layout.flex_flow = "column"`.
```

Comment:
I did notice  in the time window controllers, many take an input `orientation: Literal["horizontal", "vertical"]` - is this something you'd like added to the base class, potentially?

Right now I'm leaving it entirely in the hands of the user to modify the `.layout` after initializing the controller